lessons/modelevaluation.ipynb

```diff
@@ -0,0 +1,786 @@
+    "\n",
+    "3) The evidence is linearly sensitive to prior volume ($f$), but exponentially sensitive to goodness of fit ($L_{\\rm max} \\propto e^{-\\chi^2_{\\rm min}/2}$). It's still a likelihood, after all."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "The evidence for model $H$, $P(D\\,|\\,H)$, enables a form of Bayesian hypothesis testing: model comparison with the \"evidence ratio\" or \"odds ratio\" or \"Bayes Factor\" $R$\n",
+    "\n",
+    "$R = \\frac{P(D|H_2)}{P(D|H_1)}$\n",
+    "\n",
+    "$R$ is a *fully marginalized likelihood ratio* - which is to say that it *takes into account our uncertainty about values of the parameters of each model by integrating over all plausible values of them.*"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Notice that if your two models are equally probable _a priori_, then \n",
+    "\n",
+    "$\\frac{P(H_2)}{P(H_1} = 1$ such that $\\frac{P(H_2|D)}{P(H_1|D)} = R$\n",
+    "\n",
+    "This assumption is often not always easy to justify."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "A more practical way to interpret the Bayes factor is to note that it updates the model prior ratio into a posterior one. This means that:\n",
+    "\n",
+    "  * If you think that, despite having seen the data, your two models are *still equally probable,*\n",
+    "  \n",
+    "  * then the evidence ratio in favor of $H_2$ is _the odds that you would have had to have been willing to take against $H_2$, before seeing the data._"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "In our linear model fit example, we can compute the evidence for the linear and quadratic models, and form the odds ratio $R$.\n",
+    "\n",
+    "```\n",
+    "log Evidence for Straight Line Model: -157.2\n",
+    "log Evidence for Quadratic Model: -120.7\n",
+    "Evidence ratio in favour of the Quadratic Model:\n",
+    "  7e15 to 1\n",
+    "```\n",
+    "\n",
+    "The 26 unit difference in log evidence between the two models translates to a _huge_ odds ratio in favour of the quadratic model."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "Incidentally those data did not come from *either* a linear or quadratic model..."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "The same Jeffreys scale used to interpret the information criteria can be used to interpret evidence ratios:\n",
```

Comment:
Isn't it more applicable to the posterior ratio `P(H1|D)/P(H2|D)`?

Reply:
I think Jeffreys defined it for Bayes factors...